src/Nfts/NftsNavigator.tsx

```diff
@@ -0,0 +1,95 @@
+import {createStackNavigator} from '@react-navigation/stack'
+import React, {useState} from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet, TouchableOpacity, View} from 'react-native'
+
+import {Icon, SearchBar} from '../components'
+import {NftRoutes} from '../navigation'
+import {Nfts} from './Nfts'
+
+const Stack = createStackNavigator<NftRoutes>()
+
+export const NftsNavigator = () => {
+  const strings = useStrings()
+  const [showSearch, setShowSearch] = useState(false)
+  const [search, setSearch] = useState('')
+
+  const handleClearPress = () => {
+    setSearch('')
+  }
+
+  const handleBackPress = () => {
+    setShowSearch(false)
+    setSearch('')
+  }
+
+  return (
+    <Stack.Navigator
+      screenOptions={{
+        headerTitleContainerStyle: {width: '100%', alignItems: 'center'},
+        cardStyle: {backgroundColor: '#fff'},
+      }}
+      initialRouteName="nfts"
+    >
+      <Stack.Screen
+        name="nfts"
+        options={{
+          title: strings.title,
+          headerTitleAlign: 'center',
+          header: showSearch
+            ? () => (
```

Comment:
I don't know if its possible or necessary but maybe we can consider create a searchable list component? There are some components designed that use it so it could be an opportunity. What do you think?

Reply:
Agree or just better handling of the search, eg using context or something, so we don't need to pass the search term through props multiple layers. Is this something we can do in another PR, as this one is getting quite big?

Reply:
if a pr is behind a feature flag, it doesn't need to be a complete feature